packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -845,6 +846,55 @@ export class DetailsListBase extends React.Component<IDetailsListProps, IDetails
     return 0;
   }
 
+  public updateColumn(column: IColumn, newProps: { width?: number; newColumnIndex?: number }) {
+    const NO_COLUMNS: IColumn[] = [];
+
+    const { columns = NO_COLUMNS, selectionMode, checkboxVisibility, columnReorderOptions } = this.props;
+    const { width, newColumnIndex } = newProps;
+    const index = columns.findIndex(col => col.key === column.key);
+
+    if (width) {
+      this._onColumnResized(column, width, index!);
+    }
+
+    if (newColumnIndex) {
+      const isCheckboxColumnHidden =
+        selectionMode === SelectionMode.none || checkboxVisibility === CheckboxVisibility.hidden;
+
+      const showCheckbox = checkboxVisibility !== CheckboxVisibility.hidden;
+      const columnIndex = (showCheckbox ? 2 : 1) + index!;
+
+      const draggedIndex = isCheckboxColumnHidden ? columnIndex - 1 : columnIndex - 2;
+      const targetIndex = isCheckboxColumnHidden ? newColumnIndex - 1 : newColumnIndex - 2;
+
+      let isValidTargetIndex;
+
+      if (columnReorderOptions) {
+        const frozenColumnCountFromStart = columnReorderOptions.frozenColumnCountFromStart ?? 0;
+        const frozenColumnCountFromEnd = columnReorderOptions.frozenColumnCountFromEnd ?? 0;
+        isValidTargetIndex =
+          targetIndex >= frozenColumnCountFromStart && targetIndex < columns.length - frozenColumnCountFromEnd;
+      }
+      if (!isValidTargetIndex) {
+        return;
+      }
+
+      if (columnReorderOptions) {
```

Comment:
I think you could combine these to avoid the mid-block return?
```suggestion
      if (isValidTargetIndex && columnReorderOptions) {
```